src/IIIFPresentation/API/Features/Manifest/DlcsManifestCoordinator.cs

```diff
@@ -196,19 +198,30 @@ private static List<JObject> GetAssetJObjectList(WriteManifestRequest request) =
                     {
                         dlcsInteractionRequest.Asset[AssetProperties.Manifests] = null;
                     }
-
                     break;
                 }
                 case IngestType.ManifestId:
                     dlcsInteractionRequest.Asset[AssetProperties.Manifests] = new JArray(new List<string> { manifestId });
                     break;
             }
+
+            if (!assets.TryAdd(dlcsInteractionRequest.AssetId, dlcsInteractionRequest.Asset))
+            {
+                var assetInDictionary = assets[dlcsInteractionRequest.AssetId];
```

Comment:
Debug log or something? _"Asset xxx has been specified multiple times, validating they match"_. 

Would help explain what's happening in this codeblock and also help with debugging.